### DISCUSSION TITLE
Hi how to query using SQL lang?

### DISCUSSION THREAD

**Discussion by donPuerto** [supabase/supabase#2792] [Questions] [answered]
Hi I need to use joins? how can i implement this in supabase?

**Comment by steve-chavez** [comment, marked as answer]
Hey @donPuerto,

Raw SQL can be done by:

- Creating SQL views and calling them with [select](https://supabase.io/docs/reference/javascript/select).
- Creating SQL functions and calling them with [rpc](https://supabase.io/docs/reference/javascript/rpc).

>  I need to use joins?

Simple left joins can be done purely in js-side, see [here](https://supabase.io/docs/reference/javascript/select#query-foreign-tables). For more complex queries you'll need functions or views.
